src/app/meshes/sources.ts

```diff
@@ -1,18 +1,51 @@
 import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
 import type KumaApi from '@/services/kuma-api/KumaApi'
 import type {
+  PaginatedApiListResponse as CollectionResponse,
+} from '@/types/api.d'
+import type {
+  Mesh,
   MeshInsight,
 } from '@/types/index.d'
 
 type MeshParams = {
   mesh: string
 }
+type PaginationParams = {
+  size: number
+  page: number
+}
+
+type Closeable = {close: () => void}
+
+export type MeshSource = DataSourceResponse<Mesh>
+export type MeshCollection = CollectionResponse<Mesh>
+export type MeshCollectionSource = DataSourceResponse<MeshCollection>
+
 export type MeshInsightSource = DataSourceResponse<MeshInsight>
+
 export const sources = (api: KumaApi) => {
   return {
-    '/:mesh/insights': async (params: MeshParams, source: {close: () => void}) => {
+    '/:mesh/meshes': async (params: MeshParams & PaginationParams, source: Closeable) => {
       source.close()
-      return api.getMeshInsights({ name: params.mesh })
+      const offset = params.size * params.page
```

Comment:
**Issue**: This needs to subtract one page count before multiplying as the first offset must be 0 (`size * (1 - 1)`).

```suggestion
      const offset = params.size * (params.page - 1)
```